pontus/week2/Ass3/app.js

```diff
@@ -0,0 +1,238 @@
+			let messages = this.state.messages.filter( message => message.listID === list.listID );
+
+			return <NamedList 
+				listID={list.listID} 
+				listName={list.listName} 
+				messages={messages} 
+				createMsg={this.createMsg.bind(this)} 
+				archiveMsg={this.archiveMsg.bind(this)} 
+				listProperties={listProperties}
+				moveMsg={this.moveMsg.bind(this)} 
+				deleteMsg={this.deleteMsg.bind(this)} 
+			/>; 
+		});
+
+		return (
+			<div>
+				<h3>Create New List</h3>
+				<input type="text" ref='inputField' onKeyDown={this.createList} placeholder="title" />
+				<button onClick={this.createList}>Create new list</button>
+				{lists}
+			</div>
+		);
+	}
+}
+
+class NamedList extends React.Component { 
+	constructor(props) { 
+		super(props);
+
+		this.state = { 
+			charCount: 200,
+			isError: false,
+			errorMsg: "",
+			displayChars: false
+		};
+
+		this.createMsg = (e) => { 
+			if (e.keyCode == 13 || e.which == 13 || e.type == 'click') { 
+				let text = this.refs.inputField.value;
+				this.props.createMsg(this.props.listID, text);
+				this.refs.inputField.value = "";
+				this.refs.inputField.focus();
+			}
+		}
+	}
+
+	displayCharCount() { this.setState({ displayChars: !this.state.displayChars }); }
+
+	charValidation() { 
+		let charCount = 200 - this.refs.inputField.value.length;
+
+		if (charCount < 0) this.setState({ isError: true, errorMsg: "out of characters." });
+		else this.setState({ isError: false, errorMsg: "" }) 
+
+		this.setState({ charCount });
+	}
+
+ 	render() {
+		let archivedMessages = this.props.messages.filter( message => message.isArchived );
+
+		let allMessages = this.props.messages
+			.filter( message =>  !message.isArchived )
+			.concat(archivedMessages);
+
+		let messages = allMessages.map( (message) => {
+			return <Message 
+				listID={message.listID} 
+				messageID={message.messageID} 
+				text={message.text} 
+				isArchived={message.isArchived} 
+				listProperties={this.props.listProperties}
+				listName={this.props.listName}
+				archiveMsg={this.props.archiveMsg} 
+				moveMsg={this.props.moveMsg}
+				deleteMsg={this.props.deleteMsg}
+			/> ;
+		});
+
+		let errorMsg =  (this.state.isError) ? { color: 'red' } : { color: 'black'};
+		let btnState = (this.state.isError) ? true : false;
+		let displayChars = (this.state.displayChars) ? {display: 'inline'} : {display: 'none' };
+
+		return (
+			<div>
+				<h2>{this.props.listName}</h2>
+				<ul>
+					{messages}
+				</ul> 
+				<input onBlur={this.displayCharCount.bind(this)} onKeyDown={this.createMsg} onFocus={this.displayCharCount.bind(this)} onKeyUp={this.charValidation.bind(this)} ref='inputField' type="text" />
+				<button disabled={btnState} onClick={this.createMsg}>Create New Message</button>
+				<br/>
+				<div style={errorMsg}>
+					<span style={displayChars}> {this.state.charCount} </span>
+					<span>{this.state.errorMsg}</span>
+				</div>
+			</div>
+		);
+	}
+}
+
+NamedList.propTypes = { 
```

Comment:
you should be able to do this inside the constructor (like you set the initial state)